File: Algorithms/Array/56. Merge Intervals.py
# Definition for an interval.
# class Interval(object):
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution(object):
    def merge(self, intervals):
		if(not intervals):
			return list()
		intervals = sorted(intervals, key = lambda x: x.start)
		res = [intervals[0]]
		for interval in intervals[1:]:
			cur = res[-1]
			if(cur.end < interval.start):
				res.append(interval)
			elif(cur.end < interval.end):
				cur.end = interval.end
		return res

File: Algorithms/Array/56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
		if(intervals.empty()) return vector<vector<int>>();
		sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){return a[0] < b[0];});
		vector<vector<int>> ans;
		ans.emplace_back(intervals.front());
		for(int i = 1; i < intervals.size(); i++)
		{
			if(intervals[i][0] > ans.back()[1])
			{
				ans.emplace_back(intervals[i]);
			}
			else
			{
				ans.back()[1] = max(intervals[i][1], ans.back()[1]);
			}
		}
		return ans;
    }
};
